libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,218 @@
+        """
+        Get node topology
+        :param header: header for api authentication
+        :cluster_id: id for the cluster
+        :node_id: id for particular node in cluster
+        :return: response
+        """
+        self.log.info("Get node topology request....")
+        uri_param = '/nodes'
+        if node_id is not None:
+            uri_param = uri_param + '/' + node_id
+        self.log.info("node topology endpoint: %s", uri_param)
+        self.log.info("Logging query parameter for node topology: %s", uri_param)
+        response = self.get_cluster_topology(auth_header, cluster_id, uri_param)
+        return response
+
+    #Function Not Ready
+    def verify_nodes(self, node_id=None, expected_response=HTTPStatus.OK):
+        """
+        Verify number of nodes and node names
+        """
+        resp = self.get_node_topology(node_id)
+        solution_yaml = self.get_solution_yaml()
+        result = resp.status_code == expected_response
+        if result:
+            get_response = resp.json()
+            self.log.info("Verify node names")
+            self.log.info("Verify number of nodes in resp and solution yaml")
+            nodes = len(get_response["topology"]["cluster"]["nodes"])
+            num_nodes = len(solution_yaml["solution"]["storage_set"]["nodes"])
+            if num_nodes != nodes:
+                self.log.error("Actual and expected response for number of nodes didnt match")
+                result = False
+        else:
+            self.log.error("Status code check failed.")
+        return result
+
+    def verify_storage_set(self, cluster_id: str, storage_set_id: str = None,
+                                          expected_response=HTTPStatus.OK):
+        """
+        Verify storage set details
+        """
+        resp_dix = ""
+        err_msg = ""
+        resp = self.get_storage_topology(cluster_id, storage_set_id = storage_set_id)
+        solution_yaml = self.get_solution_yaml()
+        result = resp.status_code == expected_response
+        if result:
+            get_response = resp.json()
+            self.log.info("Verifying sns and dix values in resp and solution yaml")
+            storage_set_dix = get_response[
+                "topology"]["cluster"][0]["storage_set"]["durability"]["dix"]
+            resp_dix = resp_dix.join([str(storage_set_dix["data"]), str(storage_set_dix["parity"]),
+                                      str(storage_set_dix["spare"])])
+            input_dix = solution_yaml["solution"]["storage_set"]["durability"]["dix"]
+            
+            storage_set_sns = get_response[
+                "topology"]["cluster"][0]["storage_set"]["durability"]["sns"]
+            resp_sns = resp_sns.join([str(storage_set_sns["data"]), str(storage_set_sns["parity"]),
+                                      str(storage_set_sns["spare"])])
+            input_sns = solution_yaml["solution"]["storage_set"]["durability"]["sns"]
```

Comment:
Overall comment on verification:
1. Need separate IF for sns and DIX checks
2. Proper logging for the checks
3. Error message should be concatenated.
4.

Reply:
added